mm2src/coins/z_coin.rs

```diff
@@ -1150,15 +1162,49 @@ impl MarketCoinOps for ZCoin {
         ))
     }
 
+    /// Calculates the wallet balance, divided into spendable and unspendable portions.
+    /// Unspendable balance consists of notes that are locked in the wallet.
     fn my_balance(&self) -> BalanceFut<CoinBalance> {
         let coin = self.clone();
         let fut = async move {
-            let sat = coin
-                .my_balance_sat()
+            let locked_notes = coin
+                .z_fields
+                .locked_notes_db
+                .load_all_notes()
+                .await
+                .mm_err(|e| BalanceError::WalletStorageError(e.to_string()))?
+                .iter()
+                .map(|n| n.rseed.clone())
+                .collect::<HashSet<_>>();
+
+            let wallet_notes = coin
+                .get_wallet_notes()
                 .await
-                .mm_err(|e| BalanceError::WalletStorageError(e.to_string()))?;
-            Ok(CoinBalance::new(big_decimal_from_sat_unsigned(sat, coin.decimals())))
+                .map_err(|err| BalanceError::WalletStorageError(err.to_string()))?;
+
+            let (spendable_amount, unspendable_amount) =
+                wallet_notes
+                    .into_iter()
+                    .fold((Amount::zero(), Amount::zero()), |mut acc, n| {
+                        if locked_notes.contains(&rseed_to_string(&n.rseed)) {
+                            acc.1 += n.note_value;
```

Comment:
Technically this is not exactly the 'unspendable_amount' if we understand it as unconfirmed amount.
Here we calculate 'unspendable_amount' as sum of 'locked notes', but they are actually 'spent notes' for which the spending tx is not yet mined (so spent notes eventually would be removed from the balance but in place the change note would be added to it).
That is, to calculate the exact 'unspendable_amount' we should store also the change note of the spending tx and use its value in unspendable_amount (rather than spent notes value).

Maybe we could make this as TODO and add this in next PR, 
cause this PR gets too big and overcomplicated.

Reply:
Yes, now we store change notes as well and calculate my unspendable balance correctly, using change notes.